torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -206,47 +206,54 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::sendImpl(
   // will not keep the Python reference alive even if it originally comes from
   // the C++ land. Hence, we have to explicitly use the ``workerId`` in the C++
   // land.
-  enqueueSend(SendWork(workerIds_[to.id_], std::move(message)));
+  enqueueSend(SendWork(workerIds_[to.id_], std::move(message)), future);
   return future;
 }
 
-void ProcessGroupAgent::enqueueSend(SendWork work) {
+void ProcessGroupAgent::enqueueSend(
+    SendWork work,
+    std::shared_ptr<FutureMessage> future) {
```

Comment:
This future can be retrieved using `futures_[work.message_.id()]`, do we still need to pass in the future as an arg?